Sources/NIOPerformanceTester/WebSocketFrameDecoderBenchmark.swift

```diff
@@ -15,36 +15,86 @@
 import NIO
 import NIOWebSocket
 
-
 final class WebSocketFrameDecoderBenchmark {
 
     private let channel: EmbeddedChannel
     private let runCount: Int
-    private var data: ByteBuffer
+    private let dataSize: Int
+    private let maskingKey: WebSocketMaskingKey?
+    private var data: ByteBuffer!
 
-    init(dataSize: Int, runCount: Int) {
+    init(dataSize: Int, runCount: Int, maskingKey: WebSocketMaskingKey? = nil) {
         self.channel = EmbeddedChannel()
-        self.data = ByteBufferAllocator().buffer(capacity: dataSize)
+        self.dataSize = dataSize
+        self.maskingKey = maskingKey
         self.runCount = runCount
     }
 }
 
 extension WebSocketFrameDecoderBenchmark: Benchmark {
 
     func setUp() throws {
-        self.data.writeBytes([0x81, 0x00]) // empty websocket
+        self.data = ByteBufferAllocator().buffer(size: dataSize, maskingKey: maskingKey)
         try self.channel.pipeline.addHandler(ByteToMessageHandler(WebSocketFrameDecoder())).wait()
     }
 
     func tearDown() {
         _ = try! self.channel.finish()
     }
 
-    func run() throws  -> Int{
-        for _ in 0..<runCount {
+    func run() throws -> Int {
+        if let maskingKey = self.maskingKey {
+            return try self.run(with: maskingKey)
+        } else {
+            return try self.runWithoutMaskingKey()
+        }
+    }
+
+    private func runWithoutMaskingKey() throws -> Int {
+        for _ in 0..<self.runCount {
+            try self.channel.writeInbound(self.data)
+            let _: WebSocketFrame? =  try self.channel.readInbound()
+        }
+        return 1
+    }
+
+    private func run(with maskingKey: WebSocketMaskingKey) throws -> Int {
+        for _ in 0..<self.runCount {
+            self.data.webSocketUnmask(maskingKey)
```

Comment:
We actually don't need to call this function: the frame decoder should correctly handle this for us. So we don't need the two separate `run` functions, they can have the same implementation.